from PyQt5 import uic
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QWidget, QApplication

QApplication.setAttribute(Qt.AA_EnableHighDpiScaling, True)

# Class that contains the widget GUI that displays a list with help related to the program
class HelpWidget(QWidget):
    def __init__(self):
        super().__init__()
        uic.loadUi("QHelp.ui", self)

    # Changes the text displayed depending on the option chosen on the list
    def help_changed(self, help):

        if help == "Help":
            self.label_1.setText("Help will appear here!")
            self.label_12.setText("")
            self.label_2.setText("")
            self.label_3.setText("")
            self.label_4.setText("")
            self.label_5.setText("")
            self.label_6.setText("")
            self.label_7.setText("")
            self.label_8.setText("")
            self.label_9.setText("")
            self.label_10.setText("")
            self.label_11.setText("")

        elif help == "Add Prisoner":
            self.label_1.setText("")
            self.label_12.setText("Add Prisoner")
            self.label_2.setText("This option will add a new ")
            self.label_3.setText("prisoner and store it into ")
            self.label_4.setText("the database of the program")
            self.label_5.setText("")
            self.label_6.setText("Make sure all of the values and fields")
            self.label_7.setText("are entered in their corresponding formats")
            self.label_8.setText("")
            self.label_9.setText("")
            self.label_10.setText("")
            self.label_11.setText("")

        elif help == "Height format":
            self.label_1.setText("")
            self.label_12.setText("Height Format")
            self.label_2.setText("The height must be entered")
            self.label_3.setText("as a decimal number (as feet[ft]")
            self.label_4.setText("and inches[in]), in the format below:")
            self.label_5.setText("")
            self.label_6.setText("Example: ")
            self.label_7.setText("5.07 = 5ft 7in (5'7\")")
            self.label_8.setText("5.1 or 5.10 = 5ft 10in (5' 10\")")
            self.label_9.setText("5.11 = 5ft 11in (5'11\")")
            self.label_10.setText("")
            self.label_11.setText("The height entered must be valid")

        elif help == "Dates format":
            self.label_1.setText("")
            self.label_12.setText("Dates Format")
            self.label_2.setText("The dates must be entered in")
            self.label_3.setText("the following format (as integers):")
            self.label_4.setText("")
            self.label_5.setText("YYYY-MM-DD")
            self.label_6.setText("YYYY - year")
            self.label_7.setText("MM - month")
            self.label_8.setText("DD - day")
            self.label_9.setText("")
            self.label_10.setText("Make sure you enter a valid date")
            self.label_11.setText("")

        elif help == "Hair/Eye Color":
            self.label_1.setText("")
            self.label_12.setText("Hair/Eye Color")
            self.label_2.setText("Can be entered as a regular string")
            self.label_3.setText("")
            self.label_4.setText("However, you can only enter the name")
            self.label_5.setText("of the color (it will not accept numbers)")
            self.label_6.setText("")
            self.label_7.setText("")
            self.label_8.setText("")
            self.label_9.setText("")
            self.label_10.setText("")
            self.label_11.setText("")

        elif help == "Filling the fields":
            self.label_1.setText("")
            self.label_12.setText("Filling the fields")
            self.label_2.setText("When adding a new prisoner,")
            self.label_3.setText("make sure all of the field boxes")
            self.label_4.setText("are filled")
            self.label_5.setText("")
            self.label_6.setText("The only exception to this is")
            self.label_7.setText("the \"date to be liberated\" field box")
            self.label_8.setText("(refer to \"Date to be Liberated format\"")
            self.label_9.setText("in help for more info.)")
            self.label_10.setText("")
            self.label_11.setText("")

        elif help == "Date to be Liberated format":
            self.label_1.setText("Date to be")
            self.label_12.setText("Liberated format")
            self.label_2.setText("When adding a new prisoner, the only ")
            self.label_3.setText("field box that doesn't have to be filled")
            self.label_4.setText("is the \"date to be liberated\" field box")
            self.label_5.setText("")
            self.label_6.setText("This field box accepts the following inputs:")
            self.label_7.setText("If there's a liberation date:")
            self.label_8.setText("-a date in the \"Dates format\"")
            self.label_9.setText("If there's no liberation date:")
            self.label_10.setText("-Leave field box empty")
            self.label_11.setText("-Write \"None\" or \"none\"")

        elif help == "Add Image":
            self.label_1.setText("")
            self.label_12.setText("Add Image")
            self.label_2.setText("Press this button to add an")
            self.label_3.setText("image of the prisoner")
            self.label_4.setText("")
            self.label_5.setText("You will be prompted to add an image from")
            self.label_6.setText("a file directory from your device. Then, the")
            self.label_7.setText("directory of said image will be stored in the")
            self.label_8.setText("database of the program for later use. The")
            self.label_9.setText("image inserted (and its name) will be visible")
            self.label_10.setText("while adding, modifying, and viewing a prisoner.")
            self.label_11.setText("Every prisoner requires an image (.png or .jpg)")

        elif help == "Maximum character amount of each field":
            self.label_1.setText("The maximum character ")
            self.label_12.setText("amount for each field is:")
            self.label_2.setText("Last name - 30")
            self.label_3.setText("First name - 30")
            self.label_4.setText("Photo(file directory) - 150")
            self.label_5.setText("Height - 2 (numbers before decimal point), ")
            self.label_6.setText("         2 (numbers after decimal point)")
            self.label_7.setText("Hair color - 15")
            self.label_8.setText("Eye color - 15")
            self.label_9.setText("Place of Birth - 30")
            self.label_10.setText("Sentence - 255")
            self.label_11.setText("Dates - 10 (following the \"Dates format\")")

        elif help == "Modify Prisoner":
            self.label_1.setText("")
            self.label_12.setText("Modify Prisoner")
            self.label_2.setText("This option will ask the user to")
            self.label_3.setText("enter the id of the prisoner being")
            self.label_4.setText("modified, and will display the current")
            self.label_5.setText("information of the prisoner, which can")
            self.label_6.setText("then be changed as needed")
            self.label_7.setText("")
            self.label_8.setText("The new information added must follow")
            self.label_9.setText("the same field and format rules as")
            self.label_10.setText("the ones in \"Add Prisoner\"")
            self.label_11.setText("")

        elif help == "Prisoner ID":
            self.label_1.setText("")
            self.label_12.setText("Prisoner ID")
            self.label_2.setText("The prisoner's ID will be visible while")
            self.label_3.setText("modifying and viewing a prisoner")
            self.label_4.setText("")
            self.label_5.setText("However, it's the only attribute that cannot")
            self.label_6.setText("be changed and, as such, will always be")
            self.label_7.setText("associated with said prisoner")
            self.label_8.setText("")
            self.label_9.setText("Thanks to being an unique value, the ID is")
            self.label_10.setText("used to individually search for a prisoner")
            self.label_11.setText("")

        elif help == "Database Error":
            self.label_1.setText("")
            self.label_12.setText("Database Error")
            self.label_2.setText("While adding or modifying a prisoner, one")
            self.label_3.setText("might run into this error at certain times.")
            self.label_4.setText("This can happen when pressing \"modify\" when")
            self.label_5.setText("modifying a prisoner, but not changing any")
            self.label_6.setText("values. This can be prevented by pressing")
            self.label_7.setText("\"close\". Another cause may be entering an")
            self.label_8.setText("incorrect date format or surpassing the max")
            self.label_9.setText("character limit of a field. To fix this, refer")
            self.label_10.setText("to \"Dates format\" and \"Maximum character")
            self.label_11.setText("amount of each field\" respectively.")

        elif help == "Search/View Prisoner":
            self.label_1.setText("")
            self.label_12.setText("Search/View Prisoner")
            self.label_2.setText("This option will ask the user to")
            self.label_3.setText("enter the id of the prisoner being")
            self.label_4.setText("searched for, and will display the current")
            self.label_5.setText("information of the prisoner")
            self.label_6.setText("")
            self.label_7.setText("The prisoner's information in this option")
            self.label_8.setText("can only be viewed and, as such, cannot be")
            self.label_9.setText("modified")
            self.label_10.setText("")
            self.label_11.setText("")

        elif help == "Prisoners List":
            self.label_1.setText("")
            self.label_12.setText("Prisoners List")
            self.label_2.setText("This option will display a list of all")
            self.label_3.setText("of the prisoner's information, for each")
            self.label_4.setText("and every prisoner in the database of")
            self.label_5.setText("the program")
            self.label_6.setText("")
            self.label_7.setText("The prisoner's information in this option")
            self.label_8.setText("can only be viewed and, as such, cannot be")
            self.label_9.setText("modified")
            self.label_10.setText("The table will update upon refreshing")
            self.label_11.setText("the program")

        elif help == "About":
            self.label_1.setText("")
            self.label_12.setText("About")
            self.label_2.setText("This option displays the name")
            self.label_3.setText("and logo of the wardens in")
            self.label_4.setText("charge of creating, maintaining,")
            self.label_5.setText("and looking over this prison")
            self.label_6.setText("")
            self.label_7.setText("They also happen to be the")
            self.label_8.setText("creators of this very same")
            self.label_9.setText("program and database")
            self.label_10.setText("")
            self.label_11.setText("")

        """
        elif help == "":
            self.label_1.setText("")
            self.label_12.setText("")
            self.label_2.setText("")
            self.label_3.setText("")
            self.label_4.setText("")
            self.label_5.setText("")
            self.label_6.setText("")
            self.label_7.setText("")
            self.label_8.setText("")
            self.label_9.setText("")
            self.label_10.setText("")
            self.label_11.setText("")
        """